### PR TITLE
Refactor bootstrap w.r.p new Fix design

### PR DESCRIPTION
**Pull request by yhdengh**

### CODE REVIEW

File: src/fix-driver/compile.wat [hunk 1]
```diff
@@ -48,8 +50,8 @@
     (local.set $resource_limits (table.get $ro_table_0 (i32.const 0)))
     ;; encode[2] is input.wasm
     (local.set $input (table.get $ro_table_0 (i32.const 2)))
-    (call $get_value_type (local.get $input))
-    (i32.eq (i32.const 2))
+    (call $is_blob (local.get $input))
+    (i32.eq (i32.const 1))
```
Comment (Akshay-Srivatsan) [inline src/fix-driver/compile.wat:54]: is this needed? I thought Wasm uses C-style truthiness already

File: src/fix-driver/compile.wat [hunk 2]
```diff
@@ -156,15 +159,17 @@
     (table.set $rw_table_2 (i32.const 2) (local.get $curried_cc))
     (table.set $rw_table_2 (i32.const 3) (local.get $c_files))
     (call $create_tree_rw_table_2 (i32.const 4))
-    (call $create_thunk)
+    (call $create_application_thunk)
+    (call $create_strict_encode)
```
Comment (Akshay-Srivatsan) [inline src/fix-driver/compile.wat:163]: one of the few times I'm glad that Wasm is a stack machine

File: .gitmodules [hunk 3]
```diff
@@ -0,0 +1,3 @@
+[submodule "fix"]
```
Comment (Akshay-Srivatsan) [inline .gitmodules:1]: is having it as a submodule going to make releases annoying? we could also use a subtree if we don't want to pin a specific version

File: serialize.sh [hunk 4]
```diff
@@ -4,4 +4,9 @@ SRC=`realpath ${SRC_REL}`
 FIX=${SRC}/.fix
 
 mkdir -p ${FIX}
+mkdir -p ${FIX}/data
```
Comment (Akshay-Srivatsan) [inline serialize.sh:7]: the `Repository` class should probably be able to do this itself

File: src/wasm2c/initcomposer.cc [hunk 5]
```diff
@@ -199,11 +199,11 @@ class InitComposer
 
 void InitComposer::write_attach_tree()
 {
-  result_ << "extern void fixpoint_attach_tree(__m256i, wasm_rt_externref_table_t*);" << endl;
+  result_ << "extern void fixpoint_attach_tree(u8x32, wasm_rt_externref_table_t*);" << endl;
```
Comment (Akshay-Srivatsan) [inline src/wasm2c/initcomposer.cc:202]: Where does this typedef come from?

**Review by Akshay-Srivatsan — APPROVED** [review]
A few questions/comments, but otherwise looks good

File: serialize.sh (hunk 4 above)
Comment (yhdengh) [reply serialize.sh:7]: I think the `Repository` class is not creating `{FIX}/data` if it does not exist, but maybe `Repository` should be the one handling creating these directories?

File: src/wasm2c/initcomposer.cc (hunk 5 above)
Comment (yhdengh) [reply src/wasm2c/initcomposer.cc:202]: It's in `wasm_rt.h`

File: .gitmodules (hunk 3 above)
Comment (yhdengh) [reply .gitmodules:1]: But fix has submodules...We could probably make it work by copying `fix/.gitmodules` to `bootstrap` to make it work but then `bootstrap` still has submodules.

**Pull request merged**